src/app/new-page/component/content-animation/content-animation.component.ts

```diff
@@ -0,0 +1,133 @@
+})
+export class ContentAnimationNewComponent implements OnChanges, OnDestroy {
+  // eslint-disable-next-line @angular-eslint/no-input-rename
+  @Input() item: Animation;
+
+  private _unsubscribeAll = new Subject<any>();
+  animation: Animation;
+  ready: boolean;
+  anmResource: string;
+  anmViewItem: any;
+  hasEvents: boolean;
+  dir$: Observable<string>;
+  lottieOptions: any;
+
+  constructor(private pageService: PageService) {
+    this.dir$ = this.pageService.pageDir$;
+  }
+
+  ngOnDestroy() {
+    this._unsubscribeAll.next();
+    this._unsubscribeAll.complete();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    for (const propName in changes) {
+      if (changes.hasOwnProperty(propName)) {
+        switch (propName) {
+          case 'item': {
+            if (
+              !changes['item'].previousValue ||
+              changes['item'].currentValue !== changes['item'].previousValue
+            ) {
+              this.ready = false;
+              this.animation = this.item;
+              this.init();
+            }
+          }
+        }
+      }
+    }
+  }
+
+  onAnimationClick(): void {
+    if (this.animation.events) {
+      let action = '';
+      this.animation.events.forEach((event, idx) => {
+        const value = event?.namespace
+          ? `${event.namespace}:${event.name}`
+          : event.name;
+        action += idx ? ` ${value}` : value;
+      });
+      this.pageService.formAction(action);
+    }
+  }
+
+  onAnimationCreated(anim: any) {
+    this.anmViewItem = anim;
+  }
+
+  private init(): void {
+    // TODO
+    // Need to update to e31_1 when we release a new Shared Parser.
+    const resource = {
+      name: (this.animation as any).e31_1 || ''
+    };
+    this.anmResource = this.pageService.getAnimationUrl(resource.name || '');
+    if (
+      this.anmResource === resource.name &&
+      !this.anmResource.includes('http')
+    ) {
+      this.anmResource = this.pageService.findAttachment(resource.name) || '';
+    }
+
+    if (this.anmResource) {
+      this.lottieOptions = {
+        path: this.anmResource,
+        loop: !!this.animation.loop,
+        autoplay: !!this.animation.autoPlay
+      };
+    }
+
+    this.hasEvents = !!this.animation.events;
+
+    if (!!this.animation.playListeners || !!this.animation.stopListeners) {
+      this.awaitAnimEvent();
+    }
+    this.ready = true;
+  }
+
+  private awaitAnimEvent(): void {
+    this.pageService.contentEvent$
+      .pipe(takeUntil(this._unsubscribeAll))
+      .subscribe((event) => {
+        if (!this.anmViewItem) return;
+        const playListeners = this.animation.playListeners.filter(
+          (listener) => listener.name === event
+        ).length;
+        const stopListeners = this.animation.stopListeners.filter(
+          (listener) => listener.name === event
+        ).length;
+        const events = this.animation.events.filter(
+          (e) => e.name === event
+        ).length;
```

Comment:
so you shouldn't be using the `events` property here as listeners, the `events` property is just events that are triggered when the animation is clicked on.

Reply:
Yes, this function is watching for all events, `play` `stop` and clicks on the animation. which is why I look at the `this.animation.events` as well as the `playListeners` and `stopListeners`.

Reply:
ah, in that case you shouldn't be doing this below:
```typescript
if (events) {
  this.anmViewItem.togglePause()
}
```

just because an animation can trigger events when it's clicked doesn't mean that it should start/stop playback

Reply:
Hmm, I guess there isn't a strict naming convention for the events, so I can't program if an event is a play or stop event. I will remove the event functionality